api/apps/articles/model/index.coffee

```diff
@@ -38,20 +38,24 @@ retrieve = require './retrieve'
 @save = (input, accessToken, callback) ->
   validate input, (err, input) =>
     return callback err if err
-    mergeArticleAndAuthor input, accessToken, (err, article, author, publishing) ->
-      return callback(err) if err
+    id = ObjectId (input.id or input._id)?.toString()
+    @find id.toString(), (err, article = {}) =>
+      return callback err if err
       generateKeywords article, accessToken, input, (err, article) ->
         debug err if err
-        # Merge fullscreen title with main article title
-        article.title = article.hero_section.title if article.hero_section?.type is 'fullscreen'
-        if publishing
-          onPublish article, author, accessToken, sanitizeAndSave(callback)
-        else if not publishing and not article.slugs?.length > 0
-          generateSlugs article, author, sanitizeAndSave(callback)
-        else
-          sanitizeAndSave(callback)(null, article)
+        generateArtworks article, accessToken, input, (err, article) ->
+          debug err if err
+          mergeArticleAndAuthor article, input, accessToken, (err, article, author, publishing) ->
```

Comment:
Small, but it might help readability if the arguments of these functions lined up, e.g.

````coffeescript
 generateKeywords input, article, accessToken, ->
   generateArtworks input, article, accessToken, ->
     mergeArticleAndAuthor input, article, accessToken, ->
````